aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -213,13 +242,10 @@ Tensor sparse_csr_tensor(
     c10::optional<bool> pin_memory) {
   // See [Note: hacky wrapper removal for TensorOptions]
   TensorOptions options = TensorOptions().dtype(dtype).layout(layout).device(device).pinned_memory(pin_memory);
-  std::array<int64_t, 2> size = {0, 0};
-  if (col_indices.numel() > 0) {
-    AT_DISPATCH_INDEX_TYPES(col_indices.scalar_type(), "csr_construct_check", [&] {
-      size[0] = crow_indices.numel() - 1;
-      size[1] = col_indices.max().item<index_t>() + 1;
-    });
-  }
+  // std::array<int64_t, 2> size = {0, 0};
+  auto size = DimVector(IntArrayRef(col_indices.sizes().data(), col_indices.dim() - 1));
+  size.push_back(crow_indices.size(-1) - 1);
+  size.push_back(col_indices.max().item<int64_t>() + 1);
```

Comment:
col_indices are always guaranteed to be int64_t now?

Reply:
No, but here `.max().item()` gives a Scalar and `.item<int64_t>` casts the Scalar to `int64_t`.
https://github.com/pytorch/pytorch/blob/45da320092a6e23774a9cca5d7aaeb77abce2f99/aten/src/ATen/templates/TensorMethods.cpp#L25